### ISSUE TITLE
Asking for advice on how to improve rendering times

### ISSUE THREAD

**Issue by MorisNotari** [ngx-formly/ngx-formly#3733]
**Asking for advice on how to improve rendering times**
I'm seraching for advice on how to improve rendering times for forms with hundreds of components.
The example I've set up on StackBlitz (link below) has more than a thousand components and it takes about 3 seconds to render on my dev machine:

[https://stackblitz.com/edit/stackblitz-starters-whq6az?file=src%2Fapp.component.ts](https://stackblitz.com/edit/stackblitz-starters-whq6az?file=src%2Fapp.component.ts)

That is a typical scenario for the project i'm working on and we need to cut down this times (hopefully) below 1 sec.

For more context:
* forms are usually organized in tabs or sections
* forms must be rendered in a legacy mode with absolute positioning and in a new responsive mode
* the "Label" type was created to better handle absolute positioning
* the evaluation of several expressions for a large number of components might be required.

I thought of splitting a single monolithic form in multiple forms (for instance one per page/section) to be rendered in parallel/deferred way while sharing the same model... Is this a viable solution? Any drawbacks with that approach?

Thanks

**Comment by MorisNotari** [comment]
Never mind, I was able to cut down rendering times by removing "hidden" fields definitions as suggested in [#1711](https://github.com/ngx-formly/ngx-formly/issues/1711#issuecomment-519962816) and optimizing the number of fields requested in the forms.
